src/Microsoft.IdentityModel.Tokens/ECDsaAdapter.cs

```diff
@@ -0,0 +1,401 @@
+            BCRYPT_ECDSA_PUBLIC_P256_MAGIC = 0x31534345,
+            BCRYPT_ECDSA_PUBLIC_P384_MAGIC = 0x33534345,
+            BCRYPT_ECDSA_PUBLIC_P521_MAGIC = 0x35534345,
+            BCRYPT_ECDSA_PRIVATE_P256_MAGIC = 0x32534345,
+            BCRYPT_ECDSA_PRIVATE_P384_MAGIC = 0x34534345,
+            BCRYPT_ECDSA_PRIVATE_P521_MAGIC = 0x36534345,
+        }
+
+        /// <summary>
+        /// Returns the magic value representing the curve corresponding to the curve id.
+        /// </summary>
+        /// <param name="curveId">Represents ecdsa curve -P256, P384, P512</param>
+        /// <param name="willCreateSignatures">Whether the provider will create signatures or not</param>
+        /// <returns>Uint representing the magic number</returns>
+        private uint GetMagicValue(string curveId, bool willCreateSignatures)
+        {
+            if (string.IsNullOrEmpty(curveId))
+                throw LogHelper.LogArgumentNullException(nameof(curveId));
+
+            KeyBlobMagicNumber magicNumber;
+            switch (curveId)
+            {
+                case JsonWebKeyECTypes.P256:
+                    if (willCreateSignatures)
+                        magicNumber = KeyBlobMagicNumber.BCRYPT_ECDSA_PRIVATE_P256_MAGIC;
+                    else
+                        magicNumber = KeyBlobMagicNumber.BCRYPT_ECDSA_PUBLIC_P256_MAGIC;
+                    break;
+                case JsonWebKeyECTypes.P384:
+                    if (willCreateSignatures)
+                        magicNumber = KeyBlobMagicNumber.BCRYPT_ECDSA_PRIVATE_P384_MAGIC;
+                    else
+                        magicNumber = KeyBlobMagicNumber.BCRYPT_ECDSA_PUBLIC_P384_MAGIC;
+                    break;
+                case JsonWebKeyECTypes.P512: // treat 512 as 521. 512 doesn't exist, but we released with "512" instead of "521", so don't break now.
+                case JsonWebKeyECTypes.P521:
+                    if (willCreateSignatures)
+                        magicNumber = KeyBlobMagicNumber.BCRYPT_ECDSA_PRIVATE_P521_MAGIC;
+                    else
+                        magicNumber = KeyBlobMagicNumber.BCRYPT_ECDSA_PUBLIC_P521_MAGIC;
+                    break;
+                default:
+                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10645, curveId)));
+            }
+            return (uint)magicNumber;
+        }
+
+        /// <summary>
+        /// Tests if user's runtime platform supports operations using <see cref="CngKey"/>.
+        /// </summary>
+        /// <returns>True if operations using <see cref="CngKey"/> are supported on user's runtime platform, false otherwise.</returns>
+        [MethodImpl(MethodImplOptions.NoOptimization)]
+        private bool SupportsCNGKey()
+        {
+            try
+            {
+                _ = CngKeyBlobFormat.EccPrivateBlob;
+                return true;
+            }
+            catch (PlatformNotSupportedException)
```

Comment:
Would there be any harm in catching "Exception" ?

Reply:
I guess not. Catching any exception will make the code more resilient, as discussed. Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1058/commits/2ad287831df091cb053b771d931553565efec723